src/js/AppContainer.jsx

```diff
@@ -0,0 +1,45 @@
+var React = require('react');
+var api = require('./api');
+var UserSelect = require('./UserSelect.jsx');
+var UserFiHeading = require('./UserFiHeading.jsx');
+
+var AppContainer = React.createClass({
+
+  getInitialState: function(){
+    return {
+      user: {
+        name: '',
+        institutions: []
+      }
+    }
+  },
+
+  selectCallback: function(e){
+    var user = e.target.value;
+    var self = this;
+    console.log('selecting user: ', user);
+    api.getInstitutions(user, function(institutions){
+      self.setUser({name: user, institutions: institutions});
+    });
+  },
+
+  setUser: function(user){
+    this.setState({user: user});
+  },
+
+  render: function() {
+    return (
+      <div>
+        <UserSelect callback={this.selectCallback}/>
+        <UserFiHeading institution={{}} year="2017" user={this.state.user.name}/>
+        {React.cloneElement(
```

Comment:
Yeah sorry, I'll add in a comment.

Because this component is set up by a Route, there isn't a natural way to pass in props that get sent to dynamic children except by doing this. The children and cloned and get `institutions: this.state.user.institutions` merged into whatever props they've been given.